core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/AbstractFileBatchSource.java

```diff
@@ -99,7 +99,11 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     if (config.getSchema() != null) {
       pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
     } else {
-      pipelineConfigurer.getStageConfigurer().setOutputSchema(DEFAULT_SCHEMA);
+      if (config.format.equalsIgnoreCase("text")) {
```

Comment:
this is still not correct. you need to set it to the output schema if you know it. If schema is set provided, you should have validated it already and you can set it as the output schema. If schema is not provided but format is text, you can determine the correct output schema depending on if pathfield is set. This is exactly what you're doing elsewhere in the code, when setting schema in the Configuration. If schema is not provided and format is not text, you set output schema to null.